src/Finder/LscpuFinder.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Fidry CPUCounter Config package.
+ *
+ * (c) Théo FIDRY <theo.fidry@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Fidry\CpuCoreCounter\Finder;
+
+use function sprintf;
+
+/**
+ * The number of cores.
+ *
+ * @see https://stackoverflow.com/a/23378780/5846754
+ */
+final class LscpuFinder extends ProcOpenBasedFinder
+{
+    /**
+     * @var bool
+     */
+    private $physical;
+
+    public function __construct(bool $physical = false)
+    {
+        $this->physical = $physical;
+    }
+
+    public function getCommand(): string
+    {
+        return sprintf(
+            "lscpu -p | egrep -v '^#' | %s wc -l",
```

Comment:
if this can be done without egrep & wc the better. Any additional parsing (provided the output is not too brittle), should be extracted with `::countCpuCores()`. The reasoning is that the more commands we use the higher the chance that one is not having one of those installed. Whereas parsing strings in PHP, we can do.

There should also be an example or two of how to override the data provider in the test to test more variants

Reply:
After looking around and trying lscpu in what I have available here, it seems to me that simply `-p` will be always supported even on old machines. `-J` was added like 6 years ago and I am not sure about `-p=<desciption>` so Im trying for the "most reliable" `-p`.

Reply:
I leave it up to you whether to pick `-J` or not. But I think the rest still applies: you can do the parsing of the output in PHP rather than piping and manipulating it in the command.

Reply:
Yes, I went for the `compatibility` mode (simple `-p`) which my understanding is that is maintained as BC. Author is not happy about it but for this package I think it's probably the most reliable.